tests/src/test/java/org/eclipse/hono/tests/registry/DeviceManagementIT.java

```diff
@@ -420,6 +427,243 @@ public void testDeregisterDeviceFailsForNonExistingDevice(final VertxTestContext
             .onComplete(ctx.completing());
     }
 
+    /**
+     * Tests verifying the search devices operation.
+     *
+     * @see <a href="https://www.eclipse.org/hono/docs/api/management/#/devices/searchDevicesForTenant"> 
+     *      Device Registry Management API - Search Devices</a>
+     */
+    @Nested
+    @EnabledIfSystemProperty(named = "hono.deviceregistry.supportsSearchDevices", matches = "true")
+    class SearchDevicesIT {
+        /**
+         * Verifies that a request to search devices fails with a {@value HttpURLConnection#HTTP_NOT_FOUND}
+         * when no matching devices are found.
+         *
+         * @param ctx The vert.x test context.
+         */
+        @Test
+        public void testSearchDevicesFailsWhenNoDevicesAreFound(final VertxTestContext ctx) {
+            final Device device = new Device().setEnabled(false);
+            final String filterJson = getFilterJson("/enabled", true, "eq");
+
+            registry.registerDevice(tenantId, deviceId, device)
+                    .compose(ok -> registry.searchDevices(tenantId, Optional.empty(), Optional.empty(),
+                            List.of(filterJson), List.of(), HttpURLConnection.HTTP_NOT_FOUND))
+                    .onComplete(ctx.completing());
+        }
+
+        /**
+         * Verifies that a request to search devices fails when page size is invalid.
+         *
+         * @param ctx The vert.x test context.
+         */
+        @Test
+        public void testSearchDevicesWithInvalidPageSizeFails(final VertxTestContext ctx) {
+            final int invalidPageSize = -100;
+
+            registry.registerDevice(tenantId, deviceId)
+                    .compose(ok -> registry.searchDevices(tenantId, Optional.of(invalidPageSize), Optional.empty(),
+                            List.of(), List.of(), HttpURLConnection.HTTP_BAD_REQUEST))
+                    .onComplete(ctx.completing());
+        }
+
+        /**
+         * Verifies that a request to search devices with pageSize succeeds and the result is in accordance
+         * with the specified page size.
+         *
+         * @param ctx The vert.x test context.
+         */
+        @Test
+        public void testSearchDevicesWithValidPageSizeSucceeds(final VertxTestContext ctx) {
+            final int pageSize = 1;
+
+            CompositeFuture.all(registry.registerDevice(tenantId, new Device()), registry
+                    .registerDevice(tenantId, new Device())
+                    .compose(ok -> registry.searchDevices(tenantId, Optional.of(pageSize), Optional.empty(), List.of(),
+                            List.of(), HttpURLConnection.HTTP_OK))
+                    .onComplete(ctx.succeeding(httpResponse -> {
+                        ctx.verify(() -> {
+                            final JsonObject searchDevicesResultJson = httpResponse.bodyAsJsonObject();
```

Comment:
httpResponse.bodyAsJson(SearchDeviceResult.class)?

Reply:
:+1: I am not aware of that method. I will update it.

Reply:
I have updated to use `httpResponse.bodyAsJson(SearchDeviceResult.class)` and pushed a commit for that.